syntax = "proto3";

message TransactionPayload {

    enum PayloadType {
        TYPE_UNSET = 0;
        CREATE_CLINIC = 1;
        CREATE_DOCTOR = 2;
        CREATE_PATIENT = 3;
        CREATE_CLAIM = 4;
        ASSIGN_DOCTOR = 5;
        FIRST_VISIT = 6;
        PASS_TESTS = 7;
        ATTEND_PROCEDURES = 8;
        EAT_PILLS = 9;
        NEXT_VISIT = 10;
        ADD_LAB_TEST = 11;
        ADD_PULSE = 12;
        CREATE_LAB = 13;
    }

    PayloadType payload_type = 1;

    CreateClinic create_clinic = 2;
    CreateDoctor create_doctor = 3;
    CreatePatient create_patient = 4;
    CreateClaim create_claim = 5;
    ActionOnClaim assign_doctor = 6;
    ActionOnClaim first_visit = 7;
    ActionOnClaim pass_tests = 8;
    ActionOnClaim attend_procedures = 9;
    ActionOnClaim eat_pills = 10;
    ActionOnClaim next_visit = 11;
    AddLabTest lab_test = 12;
    AddPulse pulse = 13;
    CreateLab create_lab = 14;
}

message CreateClinic {
//    string public_key = 1;
    string name = 1;
//    repeated Permission permissions = 2;
}

message CreateDoctor {
//    string public_key = 1;
    string name = 1;
    string surname = 2;
//    repeated Permission permissions = 3;
}

message CreatePatient {
    string name = 1;
    string surname = 2;
//    repeated Permission permissions = 3;
}

message CreateLab {
    string name = 1;
//    repeated Permission permissions = 2;
}

message AddLabTest {
    string height=1;
    string weight=2;
    string gender=3;
    string a_g_ratio=4;
    string albumin=5;
    string alkaline_phosphatase=6;
    string appearance=7;
    string bilirubin=8;
    string casts=9;
    string color=10;
    string event_time=11;
    string id=12;
    string client_pkey=13;
}

message AddPulse {
//    string public_key=1;
    string id=1;
    string pulse=2;
    string timestamp=3;
    string client_pkey=4;
}

message CreateClaim {
    string claim_id = 1;
    string clinic_pkey = 2;
    string patient_pkey = 3;
}

message ActionOnClaim {
    enum Event {
        ASSIGN = 0;
        FIRST_VISIT = 1;
        PASS_TEST = 2;
        PASS_PROCEDURE = 3;
        EAT_PILLS = 4;
        NEXT_VISIT = 5;
    }

    string claim_id = 1;
    string clinic_pkey = 2;
    string description = 3;
    Event event = 4;
    string event_time = 5;
}